src/i2c.rs

```diff
@@ -181,8 +181,23 @@ where
     pub fn release(self) -> (I, SDA, SCL) {
         (self.i2c, self.sda, self.scl)
     }
-
-    fn start_transfer(&mut self, addr: u8, len: usize, direction: RD_WRN_A) {
+    
+    fn check_errors(&self) -> Result<(),Error> {
+	let isr = self.i2c.isr.read();
+        if isr.berr().bit_is_set() {
+            self.i2c.icr.write(|w| w.berrcf().set_bit());
+            return Err(Error::BusError);
+        } else if isr.arlo().bit_is_set() {
+            self.i2c.icr.write(|w| w.arlocf().set_bit());
+            return Err(Error::ArbitrationLost);
+        } else if isr.nackf().bit_is_set() {
+            self.i2c.icr.write(|w| w.nackcf().set_bit());
+	    return Err(Error::Nack);
+        }
+        return Ok(());
```

Comment:
No need for return, in fact the whole if can be transformed to not include return statement.